projects/components/src/action-menu/action-menu.component.ts

```diff
@@ -145,22 +186,59 @@ export class ActionMenuComponent<R, T> {
      */
     actionStyling = ActionStyling;
 
+    /**
+     * List of actions that are marked as {@link ActionType.STATIC_FEATURED} only
+     */
+    staticFeaturedActions: ActionItem<R, T>[];
+
+    /**
+     * Actions that depend on selected entities and belong to main menu list. The returned list length is less than or
+     * equal to the configured featured count in {@link actionDisplayConfig}
+     */
+    contextualFeaturedActions: ActionItem<R, T>[];
+
+    /**
+     * All the actions that depend on selected entities
+     */
+    contextualActions: ActionItem<R, T>[];
+
+    /**
+     * List containing all the static actions. It has static featured actions in the beginning of the list followed by
+     * non-featured static actions as children of grouped action called 'vcd.cc.action.menu.all.actions'
+     */
+    staticDropdownActions: ActionItem<R, T>[] | object;
+
+    /**
+     * List of only the actions that are marked as {@link ActionType.STATIC}
+     */
+    staticActions: ActionItem<R, T>[];
+
     /**
      * Returns the actions to be shown
      */
     getAvailableActions(actions: ActionItem<R, T>[]): ActionItem<R, T>[] {
+        // Copy is required here because, we modify the children property of certain actions below
+        const copyOfActions = getDeepCopyOfActionItems(actions);
         if (!this.calculateActionsAvailability) {
-            return actions;
+            return copyOfActions;
         }
-        return actions.filter(action => {
+        return copyOfActions.filter(action => {
             const isActionAvailable = this.isActionAvailable(action);
-            if (isActionAvailable && action.children && action.children.length) {
+            if (isActionAvailable && !!action.children?.length) {
```

Comment:
nit: 
I find it easier to read if it is:
`isActionAvailable && action.children?.length > 0`
I mean I read it as English language sentence.